astropy/units/quantity.py

```diff
@@ -1639,11 +1639,18 @@ def searchsorted(self, v, *args, **kwargs):
                                self._to_own_unit(v, check_precision=False),
                                *args, **kwargs)  # avoid numpy 1.6 problem
 
-    def argmax(self, axis=None, out=None):
-        return self.view(np.ndarray).argmax(axis, out=out)
+    if NUMPY_LT_1_22:
+        def argmax(self, axis=None, out=None):
+            return self.view(np.ndarray).argmax(axis, out=out)
+
+        def argmin(self, axis=None, out=None):
+            return self.view(np.ndarray).argmin(axis, out=out)
+    else:
+        def argmax(self, axis=None, out=None, *, keepdims=False):
+            return self.view(np.ndarray).argmax(axis=axis, out=out, keepdims=keepdims)
 
-    def argmin(self, axis=None, out=None):
-        return self.view(np.ndarray).argmin(axis, out=out)
+        def argmin(self, axis=None, out=None, *, keepdims=False):
```

Comment:
Looking at the numpy docs for comparison, they use ``np._NoValue`` for an identical outcome. But just for signature matching, would that be better? 

On a related note, maybe we should add an API comparison test, so ``Quantity`` will always match `ndarray` ?

Reply:
They use `np._NoValue` in the function and I copied that initially, but that fails because when one then super's to the method, `np._NoValue` is interpreted as `True`!  This turned out to be because they use `False` in the method, so that's what I needed to do too.

Overall, it makes some sense: the function calls an `argmin` method if it exists, and for compatibility with classes that do not update, they only pass on `keepdims` if it is not equal to `np._NoValue`. For the method, there is no such problem.